ts/screens/profile/CalendarsPreferencesScreen.tsx

```diff
@@ -35,53 +19,47 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
 /**
  * Allows the user to select one of the device available Calendars
  */
-class CalendarsPreferencesScreen extends React.PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      isLoading: true
-    };
-  }
+const CalendarsPreferencesScreen = () => {
+  const dispatch = useIODispatch();
+  const [isLoading, setIsLoading] = useState(true);
 
-  private onCalendarsLoaded = () => {
-    this.setState({ isLoading: false });
-  };
+  const preferredCalendarSaveSuccessDispatch = useCallback(
+    (calendar: Calendar) =>
+      dispatch(
+        preferredCalendarSaveSuccess({
+          preferredCalendar: calendar
+        })
+      ),
+    [dispatch]
+  );
+
+  const preferredCalendarRemoveSuccessDispatch = useCallback(
+    () => dispatch(preferredCalendarRemoveSuccess()),
+    [dispatch]
+  );
 
-  public render() {
-    const { isLoading } = this.state;
-    return (
-      <RNavScreenWithLargeHeader
-        title={{
-          label: I18n.t("profile.preferences.list.preferred_calendar.title")
-        }}
-        description={I18n.t("messages.cta.reminderCalendarSelect")}
-        contextualHelpMarkdown={contextualHelpMarkdown}
-        headerActionsProp={{ showHelp: true }}
-      >
-        <LoadingSpinnerOverlay isLoading={isLoading}>
-          <CalendarsListContainer
-            onCalendarSelected={this.props.preferredCalendarSaveSuccess}
-            onCalendarRemove={this.props.preferredCalendarRemoveSuccess}
-            onCalendarsLoaded={this.onCalendarsLoaded}
-          />
-        </LoadingSpinnerOverlay>
-      </RNavScreenWithLargeHeader>
-    );
-  }
-}
+  const onCalendarsLoaded = () => {
+    setIsLoading(false);
+  };
```

Comment:
```suggestion
  const onCalendarsLoaded = useCallback(() => {
    setIsLoading(false);
  }, []);
```
Doing so, when the loading state changes, `onCalendarsLoaded` does not change. Combining this solution with a `React.memo` around `CalendarsListContainer` we'll allow us to avoid unnecessary re-renders of the `CalendarsListContainer` component. What do you think?

Reply:
yes, thanks for noticing!! i did it!